### PR TITLE
feat(default_ad_api_helpers): support goal modification for rviz

### PR DESCRIPTION
**Pull request by isamu-takagi**
## Description

Add new goal pose topic with goal modification for rviz.

## Related links

https://github.com/autowarefoundation/autoware/issues/3406

## Tests performed

Check allow_goal_modification value using `/service_log` topic.
- False by default
- True if change the rviz goal topic to `/rviz/routing/rough_goal`

## Notes for reviewers

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by kosuke55 — APPROVED** [review]
thanks! works fine!
https://user-images.githubusercontent.com/39142679/231474481-49809d88-b09e-45d2-98fe-347310afe27a.mp4

**Comment by kosuke55** [comment]
@tkhmy @yabuta @kenji-miyake 
Could you please check as a code owner?

**Comment by kenji-miyake** [comment]
@kosuke55 @isamu-takagi Could you explain some items?
- Why does the goal from RViz need to be `rough`?
- How does it behave when you give a rough goal from RViz?

I can approve this after they become clear and seem to be rational.

**Pull request merged**
